src/packet/packetlist.js

```diff
@@ -42,7 +42,7 @@ class PacketList extends Array {
               const packet = packets.newPacketFromTag(tag, allowedPackets);
               packet.packets = new PacketList();
               packet.fromStream = util.isStream(parsed.packet);
-              await packet.read(parsed.packet, streaming);
+              await packet.read(parsed.packet, parsed.tag === enums.packet.compressedData ? streaming : config);
```

Comment:
@twiss to avoid this selection here, it's probably best to add a dummy `streaming` parameter to all `read` functions, even though it's currently used by just one function...?

Reply:
Maybe, as a temporary measure, we could pass
```suggestion
              await packet.read(parsed.packet, config, streaming);
```
here, instead of the other way around? Even though it's a bit ugly and normally I'm a stickler for consistency, but maybe in the future we can get rid of the `streaming` parameter from `CompressedDataPacket#read` (this packet is a bit of an anomaly since it decompresses already inside the `read` function, but maybe we could change that at some point), and then it will just be `packet, config` for all of them.

Reply:
TBH I also don't like having `read` depend on the configuration -- the only two packets that need that are UserId (for `config.maxUseridLength`) and Signatures (for `config.knownNotations`), so we could also consider setting the relevant preferences in the constructors, even though they are not really part of the packet state either.

Reply:
Myeah. For Signature packets, we could fix this by moving the `knownNotations` checking to `verify` instead of during parsing - that might be better anyway. And maybe we could do something similar for `maxUseridLength`. But tbh I would clean this up in a different PR, and not make any functional changes other than to the config here.

Reply:
Ok 👍 I think it would be reasonable to apply `config.maxUseridLength` to any new UserID instance, not just the one parsed through `read`.